src/util/block_tools.py

```diff
@@ -377,64 +345,75 @@ def get_consecutive_blocks(
                         continue
 
                     if not is_overflow_block:
+                        cc_vdf_challenge = slot_cc_challenge
+                        # non overflow block
                         if len(finished_sub_slots) == 0:
-                            cc_input = latest_sub_block.challenge_vdf_output
                             prev_ip_iters = calculate_ip_iters(constants, ips, latest_sub_block.required_iters)
                             new_ip_iters = unfinished_block.total_iters - prev_ip_iters
-                            rc_challenge = latest_sub_block.reward_infusion_new_challenge
+                            cc_vdf_input = latest_sub_block.challenge_vdf_output
+                            rc_vdf_challenge = latest_sub_block.reward_infusion_new_challenge
                         else:
-                            cc_input = ClassgroupElement.get_default_element()
                             new_ip_iters = ip_iters
-                            rc_challenge = slot_rc_challenge
+                            cc_vdf_input = ClassgroupElement.get_default_element()
+                            rc_vdf_challenge = slot_rc_challenge
+
+                    else:
+                        # overflow block
+                        overflow_count += 1
+                        rc_vdf_challenge = finished_sub_slots[-2].reward_chain.get_hash()
+                        cc_vdf_challenge = finished_sub_slots[-2].challenge_chain.get_hash()
+                        new_ip_iters = sp_iters
+                        cc_vdf_input = ClassgroupElement.get_default_element()
+
+                    # todo factor out vdf creation
+                    cc_ip_vdf, cc_ip_proof = get_vdf_info_and_proof(
```

Comment:
The point of separating overflow and non overflow, is that the VDFs for non-overflows are created before the end of the slot, whereas the overflows are created after the end of the slot. Here you are creating VDFs for overflow blocks as well, before the end of the slot, which is incorrect